internal/pkg/gateway/commit/blockparser.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"errors"
+
+	"github.com/golang/protobuf/proto"
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	ghErrors "github.com/pkg/errors"
+)
+
+type blockParser struct {
+	Block *commonProto.Block
+}
+
+func (parser *blockParser) TransactionValidationCodes() (map[string]peerProto.TxValidationCode, error) {
+	results := make(map[string]peerProto.TxValidationCode)
+
+	envelopes := parser.Block.Data.Data
```

Comment:
This is fine so long as your only consumer is inside the peer process (there should be no way to get back a malformed block internally), but if you want to make this operable outside the peer, probably worth a nil check on `block.Data`

Reply:
This code is very specific to an implementation embedded within a peer, but if it's allowable for `block.Data` (or any other elements) to be nil once committed to the ledger, I will add unit tests and code to handle that